src/map/packets/char.cpp

```diff
@@ -29,186 +29,407 @@
 #include "status_effect_container.h"
 #include "utils/itemutils.h"
 
-CCharPacket::CCharPacket(CCharEntity* PChar, ENTITYUPDATE type, uint8 updatemask)
+// Namespace to avoid compilation units using the same structs twice with different definitions
+namespace charflags
+{
+    // PS2: (Non-defined bitfield type.)
+    struct sendflags_t
```

Comment:
I am very much a fan of just bringing in the structs as they exist in xipackets. Maybe we should talk to atom0s about making xipackets suitable as a build-time dependency and have these defined in there?